lib/computer.rb

```diff
@@ -7,24 +7,49 @@ def initialize
     end
 
     def make_move(board)
+        win = find_win_column(board)
+        block = find_block_column(board)
+        random = find_random_column(board)
+        
+        if win
+            board.add_piece(win, @piece)
+        elsif block
+            board.add_piece(block, @piece)
+        else
+            board.add_piece(random, @piece)
+        end
+    end
+
+    def find_random_column(board)
         columns = ('A'..'G').to_a
-        board.add_piece(columns.sample, @piece)
+        columns.sample
     end
 
-    # method to scan the board for 3 in a row vertically, horizontally, and diagonally of it's own piece value and for the opponent piece value
-    # 
+    def find_win_column(board)
+        columns = ('A'..'G').to_a
+        
+        columns.each do |column|
+            board_copy = Board.new
```

Comment:
I'm glad we found this through the win? portion of this test failing. It could have been a much more difficult issue to solve later down the line if we didn't know that the cells in the original Board object were being changed even in our duplicate version.